work_queue_gantt.py

```diff
@@ -21,57 +25,93 @@ def usage(status=0):
     sys.exit(status)
 
 
-def get_properties(line):
+def get_info(line):
     return json.loads(line[line.rfind('{'):])
 
 
 def parse_tasks(logfile):
     """
-    Parse the logfile as a dictionary of properties of tasks.
-    { name: properties }
+    Parse the logfile as a table.
+    | taskid | category | wait_time | start_time | end_time | info |
     """
     
-    tasks = {}
-    
+    tasks = pd.DataFrame(columns=[
+        "taskid", "category", "wait_time", "start_time", "end_time", "info"
+    ])
+    tasks.set_index("taskid")
+    # TODO: switch from nested json to flatten table. 
     with open(logfile, 'r') as log:
+        p_taskid    = re.compile(r"^[^#].*TASK (\d+) (WAITING|RUNNING|DONE) ")
+        p_info      = re.compile(r" (\{.*\})?$")
+        p_category  = re.compile(r"WAITING (\w+)")
+        p_time      = re.compile(r"^\d+")
+        min_time    = 0
+
         for line in log:
-            if line[0] == '#':
+            m = p_taskid.findall(line)
+            if not m:
                 continue
-            
-            task_occ = line.find("TASK")
-            done_occ = line.find("DONE")
-            if task_occ != -1 and done_occ != -1:
-                name = line[task_occ:done_occ].split()[1]
-                tasks[name] = get_properties(line)
-                continue
-
+            m, = m
+            taskid  = int(m[0])
```

Comment:
I suggest using named groups in the regular expression, otherwise it gets tricky when adding new groups. Further, you can get most of the information from a single expression with something like:

```python
p_task = re.compile("(?P<time>\d+) \s+ (?P<pid>\d+) \s+ TASK \s+ (?P<taskid>\d+) \s+ (?P<state>\S+) \s+ (?P<state_args>.+)", re.X)

...

m = p_task.match(line)
if m:
   print(m.group("taskid"))

...
```